pkg/controller/volume/attach_detach_controller.go

```diff
@@ -90,7 +136,95 @@ func NewAttachDetachController(
 		DeleteFunc: adc.nodeDelete,
 	})
 
-	return adc
+	if err := adc.volumePluginMgr.InitPlugins(plugins, adc); err != nil {
+		return nil, fmt.Errorf("Could not initialize volume plugins for Attach/Detach Controller: %+v", err)
+	}
+
+	adc.desiredStateOfWorld = cache.NewDesiredStateOfWorld(&adc.volumePluginMgr)
+	adc.actualStateOfWorld = cache.NewActualStateOfWorld(&adc.volumePluginMgr)
+	adc.attacherDetacher = attacherdetacher.NewAttacherDetacher(&adc.volumePluginMgr)
+	adc.reconciler = reconciler.NewReconciler(
+		reconcilerLoopPeriod,
+		reconcilerMaxSafeToDetachDuration,
+		adc.desiredStateOfWorld,
+		adc.actualStateOfWorld,
+		adc.attacherDetacher)
+
+	return adc, nil
+}
+
+type attachDetachController struct {
+	// kubeClient is the kube API client used by informers and volumehost to
+	// communicate with the API server.
+	kubeClient internalclientset.Interface
+
+	// internalPodInformer is the shared pod informer used to fetch and store
+	// pod objects from the API server. It is shared with other controllers and
+	// therefore the pod objects in its store should be treated as immutable.
+	internalPodInformer framework.SharedInformer
+
+	// selfCreatedPodInformer is true if the internalPodInformer was created
+	// during initialization and not passed in.
+	selfCreatedPodInformer bool
```

Comment:
why does this matter?  If I can't talk you into getting rid of it, at least document it.